clients.md

```diff
@@ -89,12 +91,36 @@ From Server to client:
 * `client/unregisterCapability`
 * `textDocument/publishDiagnostics`
 
-The DLS also uses some [custom messages](contributing.md#extensions-to-the-language-server-protocol).
+### Extensions to the Language Server Protocol
```

Comment:
The format of these requests/notifications should perhaps be documented in a similar way as in the LSP specification rather than referring to Rust source code. The consumer of this documentation would typically be someone implementing a language client, which would not necessarily be someone who is well-versed in Rust.